src/lib/Simplify.hs

```diff
@@ -168,6 +176,11 @@ simplifyExpr expr = case expr of
       DataCon _ _ con args -> do
         let Abs bs body = alts !! con
         extendR (newEnv bs args) $ simplifyBlock body
+      Variant types label i value -> do
+        let LabeledItems ixtypes = enumerate types
+        let index = fst $ ixtypes M.! label !! i
```

Comment:
nit: can you please parenthesize the rhs of the `$`? It is unclear whether it's `M.!` or `!!` that takes precedence